Makefile

```diff
@@ -151,12 +151,16 @@ clean-target-dir:
 	$$RELAY_PYTHON_VERSION -m venv --copies .venv
 	.venv/bin/pip install -U pip wheel
 
+.venv/python-requirements: requirements-dev.txt
 	@# Work around https://github.com/confluentinc/confluent-kafka-python/issues/1190
 	@if [ "$$(uname -sm)" = "Darwin arm64" ]; then \
 		echo "Using 'librdkafka' from homebrew to build confluent-kafka"; \
 		export C_INCLUDE_PATH="$$(brew --prefix librdkafka)/include"; \
 	fi; \
 	.venv/bin/pip install -U -r requirements-dev.txt
+	# Bump the mtime of an empty file.
+	# Make will re-run 'pip install' if the mtime on requirements-dev.txt is higher again.
+	touch .venv/python-requirements
```

Comment:
nit: traditionally this is called a "stamp" file and tends to have that in it's name, e.g. `.venv/python-requirements-stamp`.

Reply:
didn't know that, i thought i just invented something

Reply:
now i feel like a party pooper, i blame too many makefiles in my teenage years